### ISSUE TITLE
[BUG] missing space

### ISSUE THREAD

**Issue by DonaldDai** [markedjs/marked#1634]
check this preview [link](https://marked.js.org/demo/?text=-%20obj%0A%60%60%60js%0Aconst%20a%20%3D%20%7B%0A%20%20b%3A%20%7B%0A%20%20%20%20C%3A%20%22c%22%2C%0A%20%20%7D%2C%0A%20%20a%3A%20%22a%22%2C%0A%7D%3B%0A%60%60%60%0A%0A&options=%7B%0A%20%22baseUrl%22%3A%20null%2C%0A%20%22breaks%22%3A%20false%2C%0A%20%22gfm%22%3A%20true%2C%0A%20%22headerIds%22%3A%20true%2C%0A%20%22headerPrefix%22%3A%20%22%22%2C%0A%20%22highlight%22%3A%20null%2C%0A%20%22langPrefix%22%3A%20%22language-%22%2C%0A%20%22mangle%22%3A%20true%2C%0A%20%22pedantic%22%3A%20false%2C%0A%20%22sanitize%22%3A%20false%2C%0A%20%22sanitizer%22%3A%20null%2C%0A%20%22silent%22%3A%20false%2C%0A%20%22smartLists%22%3A%20false%2C%0A%20%22smartypants%22%3A%20false%2C%0A%20%22xhtml%22%3A%20false%0A%7D&version=master)

There is an indent missing in line 2.

I've seen a similar issue [#885](https://github.com/markedjs/marked/issues/885), but the issue closed with no solution presented.

**Comment by UziTech** [comment]
looks like it thinks the code is part of the list item because there is no blank line between them so it is removing two spaces before every line.

If the code is part of the list item it should be indented two spaces which would give the desired result.

````markdown
- obj
  ```js
  const a = {
    b: {
      C: "c",
    },
    a: "a",
  };
  ```
````

[demo](https://marked.js.org/demo/?text=-%20obj%0A%20%20%60%60%60js%0A%20%20const%20a%20%3D%20%7B%0A%20%20%20%20b%3A%20%7B%0A%20%20%20%20%20%20C%3A%20%22c%22%2C%0A%20%20%20%20%7D%2C%0A%20%20%20%20a%3A%20%22a%22%2C%0A%20%20%7D%3B%0A%20%20%60%60%60%0A%0A&options=%7B%0A%20%22baseUrl%22%3A%20null%2C%0A%20%22breaks%22%3A%20false%2C%0A%20%22gfm%22%3A%20true%2C%0A%20%22headerIds%22%3A%20true%2C%0A%20%22headerPrefix%22%3A%20%22%22%2C%0A%20%22highlight%22%3A%20null%2C%0A%20%22langPrefix%22%3A%20%22language-%22%2C%0A%20%22mangle%22%3A%20true%2C%0A%20%22pedantic%22%3A%20false%2C%0A%20%22sanitize%22%3A%20false%2C%0A%20%22sanitizer%22%3A%20null%2C%0A%20%22silent%22%3A%20false%2C%0A%20%22smartLists%22%3A%20false%2C%0A%20%22smartypants%22%3A%20false%2C%0A%20%22xhtml%22%3A%20false%0A%7D&version=master)

**Comment by UziTech** [comment]
This was fixed in [v4.0.17](https://github.com/markedjs/marked/releases/tag/v4.0.17)
